avocado/core/runner.py

```diff
@@ -359,6 +359,16 @@ def sigtstp_handler(signum, frame):     # pylint: disable=W0613
 
         signal.signal(signal.SIGTSTP, sigtstp_handler)
 
+        # Saving the original sys.path state
+        _syspath = sys.path[:]
+
+        # Injecting the test module into the sys.path, so user can define
+        # a method from the test module to be used in func_at_exit
+        if 'modulePath' in test_factory[1]:
+            test_path = test_factory[1]['modulePath']
+            test_module_dir = os.path.abspath(os.path.dirname(test_path))
+            sys.path.insert(0, test_module_dir)
```

Comment:
Well this only works when the module is in the modulePath and it'll always update the `sys.path`, not just when `func_at_exit` is used. I have to say I had something else in mind, which should be IMO more scalable, so please let me know what do you think.

My approach would be to use `pickle.dumps` while pickling the `func_at_exit` and adding optional `sysPath` as an entry. Therefor the test code would do:

```python
self.runner_queue.put({"func_at_exit": pickle.dumps(cleanup_function),
                                               "args": ("Hello Avocado Test Queue!",),
                                               "once": True,
                                               "sysPath": [os.path.dirname(avocado_queue_testlib.__FILE__)]}
```

The execution won't be affected and only during the `func_at_exit.run` it'd do:

```python
func, args, kwargs = item
pth = sys.path[:]
try:
    sys.path.extend(self.path)
    func = pickle.loads(func)
    ...
finally:
    sys.path = pth
```

Which would be easy to move into a different process as we only change the `sys` for the cleanup. Ideally we'd also create `self.queue.func_at_exit` which would do the `pickle` for us, but for now we could simply leave it as is and support pickled and unpickled entries until we add the API to handle it from tests.

Anyway what do you say, would this work for you? I'd prefer this solution because:

1. It modifies the `sys` only for the cleanup (easy separation, lower impact)
2. It allows injecting custom paths (which is quite useful for complex tests)
3. It's not modifying anything unless explicitly asked for (therefor safe and fast)

The cons I see are:

1. It requires `pickle` from inside tests (could be addressed by `API` to register `func_at_exit` which  we keep postponing)